### PR TITLE
feat: further Rancher provider work

### LINKED ISSUE #12: Rancher Discovery Plugin
Create the discovery plugin that queries rancher for clusters

### LINKED ISSUE #213: Default the idp-protocol if not specified
**Describe the solution you'd like:**
If the `--idp-protocol` flag isn't specified to `kconnect use` then we should default to using the first supported identity provider 

**Why do you want this feature:**

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by richardcase**
**What this PR does / why we need it**:
Adds support for Rancher

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #12 
Fixes #213

### CODE REVIEW

File: pkg/plugins/identity/rancher/activedirectory/activedirectory.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+func (p *radIdentityProvider) Usage(clusterProvider string) (string, error) {
+	return "", nil
+}
+
+func (p *radIdentityProvider) ensureLogger() {
+	if p.logger == nil {
+		p.logger = zap.S().With("provider", ProviderName)
+	}
+}
+
```
Comment (seankelly001) [inline pkg/plugins/identity/rancher/activedirectory/activedirectory.go:152]: We need to remove this?

File: pkg/plugins/identity/static/token/token.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	cs.String(tokenConfigItem, "", "the token to use for authentication")                                  //nolint:errcheck
+	cs.SetRequired(tokenConfigItem)                                                                        //nolint:errcheck
+	cs.SetSensitive(tokenConfigItem)                                                                       //nolint:errcheck
+
+	return cs, nil
+}
+
+// Usage returns a string to display for help
+func (p *staticTokenIdentityProvider) Usage(clusterProvider string) (string, error) {
+	return "", nil
```
Comment (seankelly001) [inline pkg/plugins/identity/static/token/token.go:97]: Why is this retuning an empty string?

**Review by seankelly001 — COMMENTED** [review]
Left some comments on files

File: pkg/plugins/identity/rancher/activedirectory/activedirectory.go (hunk 1 above)
Comment (richardcase) [reply pkg/plugins/identity/rancher/activedirectory/activedirectory.go:152]: We do, just removed

File: pkg/plugins/identity/static/token/token.go (hunk 2 above)
Comment (richardcase) [reply pkg/plugins/identity/static/token/token.go:97]: We probably need to revisit the usage

**Pull request merged**